lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py

```diff
@@ -491,19 +514,19 @@ def exec_module(self, **kwargs):
 
                 if not changed:
                     nsg = self.get_security_group(self.security_group['resource_group'], self.security_group['name'])
-                    if nsg and results.get('network_security_group') and results['network_security_group'].get('id') != nsg.id:
+                    if nsg and results.get('network_security_group') and results['network_security_group'].get('id').lower() != nsg.id.lower():
```

Comment:
why need lower()? suppose sdk should  return consistent format, if not, we can help to cut an issue for sdk

Reply:
I don't necessarily think it's an issue with the SDK, the Azure API itself returns whatever case was used to create the object. So if, for instance, someone were to use a lowercase name to refer to an object that had already been created in a different case then Azure would treat that request as if it referred to the original object, but Ansible would not, and will show changed every time due to the difference in case. 

Is that something the user should be aware of, and expected to handle in their plays? If that's the expectation I will remove the `lower()` calls I've inserted.